src/lib.rs

```diff
@@ -154,6 +151,19 @@ impl<T: Clone> WatchReceiver<T> {
         self.last_seen_version = lock.version;
         lock.value.clone()
     }
+
+    /// This method waits until a new value becomes available and return a clone
+    /// of it, timing out after specified duration.
+    pub fn wait_timeout(&mut self, duration: Duration) -> Option<T> {
+        let mut lock = self.shared.lock.lock();
+
+        while lock.version == self.last_seen_version {
+            lock = self.shared.on_update.wait_timeout(lock, duration)?;
+        }
```

Comment:
Every time this loop goes around, the deadline is reset. That's incorrect.

Reply:
Right. Maybe that was my hang. Will fix.